### PR TITLE
Select consonants or vowels

### PR DESCRIPTION
**Pull request by JamieZafar**

### CODE REVIEW

File: src/client/game/game.js [hunk 1]
```diff
@@ -3,33 +3,79 @@ import { connect } from 'react-redux';
 
 import AnswerInput from './answer-input';
 
-import { setLeader } from './game-actions';
+import { setLeader, getConsonantDispatch, getVowelDispatch, 
+        disableConsonantDispatch, disableVowelDispatch } from './game-actions';
 
-import { setLeaderDispatch } from '../services/cloak-service';
+import { setLeaderDispatch, setConsonantDispatch, messageGetConsonant,
```
Comment (thullSL) [inline src/client/game/game.js]: This style is ugly AF for long lists. I propose something like this for 3+ object members
```
import {
    setLeader,
    getConsonantDispatch,
    getVowelDispatch,
    disableConsonantDispatch,
    disableVowelDispatch
} from './game-actions';
```

File: src/client/game/game.js [hunk 2]
```diff
@@ -3,33 +3,79 @@ import { connect } from 'react-redux';
...
         setLeaderDispatch(this.props.setLeader);
+        setConsonantDispatch(this.props.getConsonantDispatch);
+        setVowelDispatch(this.props.getVowelDispatch);
+        setDisableConsonantDispatch(this.props.disableConsonantDispatch);
+        setDisableVowelDispatch(this.props.disableVowelDispatch);
+    }
+
+    isLeader() {
+        let userId = storageService.getUser().name;
+        if(userId === this.props.leader) {
```
Comment (thullSL) [inline src/client/game/game.js]:
```
isLeader() {
    let userId = storageService.getUser().name;
    return userId === this.props.leader;
}
```

Also why is it `getUser().name` but the variable is called `userId`. Is it a name or an ID?

File: src/client/game/game-actions.js [hunk 3]
```diff
@@ -15,10 +20,37 @@ export function leaveGame() {
 	}
 }
 
-export function setLeader(user) {
+export function setLeader(userObj) {
```
Comment (kenneth-gray) [inline src/client/game/game-actions.js]: user seemed okay - why the change?

File: src/client/game/game-reducer.js [hunk 4]
```diff
@@ -1,23 +1,58 @@
-import { START_GAME, LEAVE_GAME, SET_LEADER } from './game-actions';
+
+import { START_GAME, LEAVE_GAME, SET_LEADER, GET_CONSONANT,
```
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: This list is getting quite long I guess, so it could be changed to `import * as actionTypes from './game-actions'`. Then use `actionTypes.START_GAME` in the reducer and so on

File: src/client/game/game-reducer.js [hunk 5]
```diff
@@ -1,23 +1,58 @@
...
+            letterList.push(action.payload);
+            return Object.assign({}, state, {
+                letterList: letterList
+            });
+        }
+        case GET_VOWEL: {
+            let letterList = Array.from(state.letterList);
+            letterList.push(action.payload);
+            return Object.assign({}, state, {
+                letterList: letterList
```
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: `letterList: [...state.letterList, action.payload]` -  you can then remove the variable assignment

File: src/client/game/game.js (hunk 1 above)
Comment (kenneth-gray) [inline src/client/game/game.js]: I agree. Also, this should be changed so you just import the cloakService anyway, but that's technical debt at the moment.

File: src/client/game/game.js [hunk 6]
```diff
@@ -3,33 +3,79 @@ import { connect } from 'react-redux';
...
+import { setLeaderDispatch, setConsonantDispatch, messageGetConsonant, 
+        setVowelDispatch, messageGetVowel, setDisableConsonantDispatch,
+        setDisableVowelDispatch } from '../services/cloak-service';
+
+import storageService from '../services/storage-service';
 
 export class Game extends Component {
     componentWillMount() {
         setLeaderDispatch(this.props.setLeader);
+        setConsonantDispatch(this.props.getConsonantDispatch);
```
Comment (kenneth-gray) [inline src/client/game/game.js]: All of this set{_Something_}Dispatch is making me nervous - I don't get it. Would appreciate an in person explanation at some point

File: src/client/game/game.js (hunk 2 above)
Comment (kenneth-gray) [inline src/client/game/game.js]: ...or a leader?

File: src/client/game/game.js [hunk 7]
```diff
@@ -3,33 +3,79 @@ import { connect } from 'react-redux';
...
+            </div>
+        );
+
         return (
             <div className="col-lg-8 text-center">
-                <h3>THIS IS GAME</h3>
-                <p>{this.props.leader}</p>
+                <h3>COUNTDOWN</h3>
+                <p>Leader: {this.props.leader}</p>
+                {this.isLeader() ? letterButtons : null}
```
Comment (kenneth-gray) [inline src/client/game/game.js]: Could move the test to the letterButtons const instead?

File: src/server/letter-lists.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+const consonants = {
```
Comment (kenneth-gray) [inline src/server/letter-lists.js]: There's a lot of es6 here. Are we happy having this on the server? @thullSL?

File: src/server/cloak-server.js [hunk 9]
```diff
@@ -162,6 +177,49 @@ function setNextLeader(room) {
...
+    if(letterList.letters.length >= 9){
+        letterList.disableConsonant = true;
+        letterList.disableVowel = true;
+        user.message('disableConsonant', true);
+        user.message('disableVowel', true);
+        return;
+    }
+}
+
+function getConsonant(arg, user) {
```
Comment (kenneth-gray) [inline src/server/cloak-server.js]: Is this or anything new being tested? I know the tests have been updated, but those are for other things right?

File: src/server/letter-lists.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+const vowels = {
+	A: 15,
+	E: 21,
+	I: 13,
+	O: 13, 
+	U: 5
+}
+
+const totalWeights = {
+	consonants: 74,
```
Comment (thullSL) [inline src/server/letter-lists.js]: `Object.keys(consonants).reduce( (total, letter)=>total  + vowels[letter], 0)`

File: src/server/letter-lists.js (hunk 8 above)
Comment (thullSL) [inline src/server/letter-lists.js]: As as long as node handles it i'm fine. I think we're on 6.something so should be good http://node.green/

Comment (kenneth-gray) [inline src/server/letter-lists.js]: In the README it says v5 or above so this is fine with me too - just be careful for what is supported and isn't for v5 or update the README to require 6+.

File: src/client/game/game-actions.js (hunk 3 above)
Comment (JamieZafar) [inline src/client/game/game-actions.js]: userObj also contains some data that isn't necessarily relevant to the user, so we named it differently to make it clear that it isn't the same as a cloak 'user'

File: src/client/game/game.js (hunk 6 above)
Comment (JamieZafar) [inline src/client/game/game.js]: Sure, whenever you're available

File: src/server/letter-lists.js [hunk 11]
```diff
@@ -30,9 +30,13 @@ const vowels = {
...
 }
 
+const sumWeights = (letters) => (
+	Object.keys(letters).reduce( (total, letter) => total + letters[letter], 0)
+); 
+
 const totalWeights = {
-	consonants: 74,
-	vowels: 67
+	consonants: sumWeights(consonants),
```
Comment (thullSL) [inline src/server/letter-lists.js]: Can't believe you added them up manually before. Like some sort of non-programming peasant :disappointed:

Comment (bumblebot1) [inline src/server/letter-lists.js]: roast of the day :fire:

**Comment by JamieZafar** [comment]
:ship: dat :hankey: ?

**Comment by thullSL** [comment]
:shipit:
